tests/system/datasources_test.go

```diff
@@ -0,0 +1,57 @@
+package system
+
+import (
+	"testing"
+)
+
+func TestDataSources(t *testing.T) {
+	t.Parallel()
+
+	s := NewSystemTest(t, "cases/datasources", nil)
+	s.Terraform.Apply()
+	defer s.Terraform.Destroy()
+
+	// Compare outputs of data and resource values (resource
+	// values have the '_expected' suffix)
+	checkOutput := func(key string) {
+		log.Debugf("Checking data source vs resource output for key: %s", key)
+		datasourceOutput := s.Terraform.Output(key)
+		resourceOutput := s.Terraform.Output(key + "_expected")
+
+		if datasourceOutput != resourceOutput {
+			log.Fatalf(
+				"Data value '%s' and Resource value '%s' do not match for key: %s",
+				datasourceOutput,
+				resourceOutput,
+				key)
+		}
+	}
+
+	//check environment outputs
+	checkOutput("environment_id")
+	checkOutput("environment_size")
+	checkOutput("environment_min_count")
+	checkOutput("environment_os")
+	checkOutput("environment_ami")
+
+	//check deploy output
+	checkOutput("deploy_id")
+
+	//check load balancer outputs
+	checkOutput("load_balancer_id")
+	checkOutput("load_balancer_name")
+	checkOutput("load_balancer_environment_name")
```

Comment:
[go-ism?] I like the `checkOutput` pattern - not advocating that it needs to change or anything - but I did want to show a pattern that lot of go devs use for cases like this: using a slice of anonymous structs to create a table test:

````
outputPairs := []struct{
    DataOutputKey     string
    ResourceOutputKey string
 }{
    {
        DataOutputKey:     "environment_id",
        ResourceOutputKey: "environment_id_expected",
    },
    {
        DataOutputKey:     "service_id",
        ResourceOutputKey: "service_id_expected",
    },
...
}

for _, o := range outputPairs {
    if v, want := s.Terraform.Output(o.DataOutputKey), s.Terraform.Output(o.ResourceOutputKey); v != want {
        t.Errorf("%s: value was '%s', expected '%s'", key, v, want)
    }
}
````

Since we only need `string:string` pairs here, we could just use a `map[string]string` instead of anonymous structs, but that's obviously a special case.